aws/table_aws_wellarchitected_check_detail.go

```diff
@@ -0,0 +1,315 @@
+				Description: "Well-Architected Lens ARN associated to the check.",
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromField("CheckDetail.LensArn"),
+			},
+			{
+				Name:        "name",
+				Description: "Trusted Advisor check name.",
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromField("CheckDetail.Name"),
+			},
+			{
+				Name:        "owner_account_id",
+				Description: "An Amazon Web Services account ID.",
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromField("CheckDetail.AccountId"),
+			},
+			{
+				Name:        "pillar_id",
+				Description: "The ID used to identify a pillar, for example, security. A pillar is identified by its PillarReviewSummary$PillarId.",
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromField("CheckDetail.PillarId"),
+			},
+			{
+				Name:        "provider",
+				Description: "Provider of the check related to the best practice.",
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromField("CheckDetail.Provider"),
+			},
+			{
+				Name:        "question_id",
+				Description: "The ID of the question.",
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromField("CheckDetail.QuestionId"),
+			},
+			{
+				Name:        "reason",
+				Description: "Reason associated to the check.",
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromField("CheckDetail.Reason"),
+			},
+			{
+				Name:        "status",
+				Description: "Status associated to the check.",
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromField("CheckDetail.Status"),
+			},
+			{
+				Name:        "updated_at",
+				Description: "The date and time recorded.",
+				Type:        proto.ColumnType_TIMESTAMP,
+				Transform:   transform.FromField("CheckDetail.UpdatedAt"),
+			},
+			{
+				Name:        "workload_id",
+				Description: "The ID of the workload.",
+				Type:        proto.ColumnType_STRING,
+			},
+
+			// Steampipe standard columns
+			{
+				Name:        "title",
+				Description: resourceInterfaceDescription("title"),
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromField("Name"),
+			},
+		}),
+	}
+}
+
+type CheckDetailInfo struct {
+	types.CheckDetail
+	WorkloadId *string
+}
+
+//// LIST FUNCTION
+
+func listWellArchitectedCheckDetails(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
+
+	answerList, err := getAnswerDetailsForWorkload(ctx, d, h)
+	if err != nil {
+		plugin.Logger(ctx).Error("aws_wellarchitected_check_detail.listWellArchitectedCheckDetails", "error", err)
+		return nil, err
+	}
+
+	if len(answerList) == 0 {
+		return nil, nil
+	}
+
+	// Create session
+	svc, err := WellArchitectedClient(ctx, d)
+	if err != nil {
+		plugin.Logger(ctx).Error("aws_wellarchitected_check_detail.listWellArchitectedCheckDetails", "client_error", err)
+		return nil, err
+	}
```

Comment:
Should we move this section of code before making the `getAnswerDetailsForWorkload` function call?